RallySoftware/react-dnd
# S112460: Upgrade to work with react 15.0.1

#### What does this PR do?

Update to work with react 15

#### Any background context you want to provide?

Sombrero is updating to react 15 and this needs to follow suit.  

#### Where should the reviewer start?

See changed files.

#### How should this be manually tested?

There are no tests for this package.
`npm install` then `npm run demo` and see that they all work

#### What are the relevant tickets?

[S112839](https://rally1.rallydev.com/#/49687616960d/detail/userstory/53782239845)

#### Screenshots, if appropriate. [:link:](http://recordit.co/)

n/a

#### Definition of Done
* [x] Story __acceptance criteria__ updated if changed
* [x] Does not negatively impacting [__perceived performance__](https://docs.google.com/document/d/1j4AbGqGp_Le1SfQp3lp-pclk0ojQqZy7Cp-HbNq4i94)
* [x] Does not degrade __performance__ and meets feature performance requirements
* [x] Renders properly in [__all supported browsers__](https://help.rallydev.com/supported-web-browsers)
* [x] __Sufficiently rich data__ used during development to expose important data relationships
* [x] __Code reviewed__ by someone who didn't write it
* [x] __Automated tests__ written and passing at appropriate levels. (unit, integration, end-to-end, etc.)
* [x] No functional __regressions__ have been introduced

#### How does this PR make you feel? [:link:](http://giphy.com/categories/emotions/)

![200](https://cloud.githubusercontent.com/assets/202773/14887069/54e459ee-0d11-11e6-8ee7-4438f9c01a42.gif)
